fido/src/main/java/com/yubico/yubikit/fido/client/BasicWebAuthnClient.java

```diff
@@ -626,6 +602,50 @@ private boolean getCtapUv(String userVerification, boolean pinProvided) throws C
         }
     }
 
+    private AuthParams getAuthParams(
+            byte[] clientDataHash,
+            boolean shouldUv,
+            @Nullable char[] pin,
+            @Nullable Integer permissions,
+            @Nullable String rpId
+    ) throws ClientError, IOException, CommandException {
+        @Nullable byte[] authToken = null;
+        @Nullable byte[] authParam = null;
+        @Nullable Integer authProtocolVersion = null;
+
+        try {
+            if (pin != null) {
+                authToken = clientPin.getPinToken(pin, permissions, rpId);
+                authParam = clientPin.getPinUvAuth().authenticate(authToken, clientDataHash);
+                authProtocolVersion = clientPin.getPinUvAuth().getVersion();
+            } else if (pinConfigured) {
+                if (shouldUv && uvConfigured) {
+                    if (ClientPin.isTokenSupported(ctap.getCachedInfo())) {
+                        authToken = clientPin.getUvToken(permissions, rpId, null);
+                        authParam = clientPin.getPinUvAuth().authenticate(authToken, clientDataHash);
+                        authProtocolVersion = clientPin.getPinUvAuth().getVersion();
+                    }
+                    // no authToken is created means that internal UV is used
+                } else {
+                    // the authenticator supports pin but no PIN was provided
+                    throw new PinRequiredClientError();
+                }
+            }
+            return new AuthParams(
+                    authParam,
+                    authProtocolVersion
+            );
+
+        } finally {
+            if (authToken != null) {
+                Arrays.fill(authToken, (byte) 0);
+            }
+            if (authParam != null) {
+                Arrays.fill(authParam, (byte) 0);
+            }
```

Comment:
Instead of clearing authParam here, and making a copy in the constructor of AuthParam, wouldn't it be easier to just not clear it, and not make a copy?